test/test_node.h

```diff
@@ -92,4 +92,73 @@ TEST(layer_tuple, comma_operator_layer_tuple2) {
   EXPECT_EQ(tuple_2.layers_.at(2)->layer_type(), "ave-pool");
 }
 
+TEST(layer_tuple, chain_operator_layer_layer1) {
+  fully_connected_layer fc1(2, 2);
+  fully_connected_layer fc2(2, 2);
+  sigmoid_layer sgm(2);
+
+  fc1 << fc2 << sgm;
+  EXPECT_EQ(fc1.next_nodes()[0], &fc2);
+  EXPECT_EQ(fc2.next_nodes()[0], &sgm);
+}
+
+TEST(layer_tuple, chain_operator_layer_layer2) {
+  auto fc1 = std::make_shared<fully_connected_layer>(2, 2);
+  auto fc2 = std::make_shared<fully_connected_layer>(2, 2);
+  auto sgm = std::make_shared<sigmoid_layer>(2);
+
+  fc1 << fc2 << sgm;
+  EXPECT_EQ(fc1->next_nodes()[0], fc2.get());
+  EXPECT_EQ(fc2->next_nodes()[0], sgm.get());
+}
+
+TEST(layer_tuple, chain_operator_tuple_layer1) {
+  fully_connected_layer fc1(2, 2);
+  fully_connected_layer fc2(2, 2);
+  concat_layer conc({shape3d(2, 1, 1), shape3d(2, 1, 1)});
+  sigmoid_layer sgm(4);
+
+  (fc1, fc2) << conc << sgm;
```

Comment:
There was a check, but the order in `prev_nodes ()` vector gets jumbled sometimes. I am unable to compare pointers directly based on index.

This problem happens only on clang.